### PR TITLE
feat: adds test coverage for AWS Role Authentication

### PR DESCRIPTION
**Pull request by marcosuma**
## Description

Link to any related issue(s): https://jira.mongodb.org/browse/INTMDB-1034

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [X] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [X] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [X] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: mongodbatlas/provider_test.go [hunk 1]
```diff
@@ -159,6 +159,30 @@ func testCheckAwsEnv(tb testing.TB) {
 	}
 }
 
+func testCheckSTSAssumeRole(tb testing.TB) {
```
Comment (lantoli) [inline mongodbatlas/provider_test.go:168]: do we need all these variables in the test?

**Review by lantoli — APPROVED** [review]

File: mongodbatlas/fw_provider_authentication_test.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import (
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+func TestAccSTSAssumeRole_basic(t *testing.T) {
```
Comment (AgustinBettati) [inline mongodbatlas/fw_provider_authentication_test.go:12]: curious if for this test case we can ensure that MONGODB_ATLAS_PUBLIC_KEY and MONGODB_ATLAS_PRIVATE_KEY are not defined so we are certain that AWS authentication if being used

Comment (marcosuma) [reply mongodbatlas/fw_provider_authentication_test.go:12]: that's a great point and actually something I had in mind to test and forgot. Let me make sure of that.

File: mongodbatlas/provider_test.go (hunk 1 above)
Comment (marcosuma) [reply mongodbatlas/provider_test.go:168]: yes because that is how we generate credentials via STS Assume Role.

**Pull request merged**